### PR TITLE
Add RoundRect support

### PR DESCRIPTION
**Pull request by MohamedRejeb**
I added a new haze modifier that accepts `RoundRect` instead of `Rect` and I changed the internal implementation to use `RoundRect` only so I don't need to duplicate the code for both types.

For SkiaShader there's only one missing case which is supporting border radius with a different x and y, (ex: `CornerRadius(x = 20f, y = 60f)`), for now in that case I take the bigger value. It needs some more work to support this, but it's rarely used.

Closes: #14

### CODE REVIEW

File: haze-jetpack-compose/src/main/kotlin/dev/chrisbanes/haze/HazeNode31.kt [hunk 1]
```diff
@@ -14,15 +14,9 @@ import android.graphics.RenderNode
...
-import androidx.compose.ui.geometry.Rect
-import androidx.compose.ui.graphics.Canvas
-import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.graphics.drawscope.ContentDrawScope
-import androidx.compose.ui.graphics.drawscope.clipRect
-import androidx.compose.ui.graphics.drawscope.draw
-import androidx.compose.ui.graphics.drawscope.drawIntoCanvas
-import androidx.compose.ui.graphics.nativeCanvas
-import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.geometry.RoundRect
```
Comment (chrisbanes) [inline haze-jetpack-compose/src/main/kotlin/dev/chrisbanes/haze/HazeNode31.kt:17]: Please run `spotlessApply`. Wildcard imports should be disabled.

File: haze/src/skikoMain/kotlin/dev/chrisbanes/haze/HazeNode.kt [hunk 2]
```diff
@@ -34,19 +36,38 @@ private const val SHADER_SKSL = """
...
+  uniform vec2 bottomRightCornerRadius;
+  uniform vec2 bottomLeftCornerRadius;
   uniform vec4 color;
   uniform float colorShift;
 
   // https://www.iquilezles.org/www/articles/distfunctions2d/distfunctions2d.htm
-  float boxSDF(vec2 position, vec2 box) {
-      vec2 q = abs(position) - box;
-      return length(max(q,0.0)) + min(max(q.x,q.y),0.0);
+  float boxSDF(vec2 position, vec2 box, vec2 radius) {
```
Comment (chrisbanes) [inline haze/src/skikoMain/kotlin/dev/chrisbanes/haze/HazeNode.kt:47]: You can use the `sdRoundedBox` function from https://iquilezles.org/articles/distfunctions2d/

File: haze/src/skikoMain/kotlin/dev/chrisbanes/haze/HazeNode.kt [hunk 3]
```diff
@@ -34,19 +36,38 @@ private const val SHADER_SKSL = """
...
+    vec2 q = abs(position) - box + vec2(cornerRadius);
+    return length(max(q,0.0)) + min(max(q.x,q.y),0.0) - cornerRadius;
   }
 
   vec4 main(vec2 coord) {
     vec2 shiftRect = (rectangle.zw - rectangle.xy) / 2.0;
     vec2 shiftCoord = coord - rectangle.xy;
-    float distanceToClosestEdge = boxSDF(shiftCoord - shiftRect, shiftRect);
+    vec2 cornerRadius;
+    if (shiftCoord.x > shiftRect.x) {
```
Comment (chrisbanes) [inline haze/src/skikoMain/kotlin/dev/chrisbanes/haze/HazeNode.kt:57]: This can all be simplified with the `sdRoundedBox` function above.

**Review by chrisbanes — COMMENTED** [review]
Looks like a great start. Unfortunately I made some changes in this code in #17 so it will need a rebase.

File: haze-jetpack-compose/src/main/kotlin/dev/chrisbanes/haze/HazeNode31.kt (hunk 1 above)
Comment (MohamedRejeb) [reply haze-jetpack-compose/src/main/kotlin/dev/chrisbanes/haze/HazeNode31.kt:17]: Done!

File: haze/src/skikoMain/kotlin/dev/chrisbanes/haze/HazeNode.kt (hunk 3 above)
Comment (MohamedRejeb) [reply haze/src/skikoMain/kotlin/dev/chrisbanes/haze/HazeNode.kt]: Done, It's much simpler now.

**Review by chrisbanes — APPROVED** [review]
Looks good, thanks!

**Pull request merged**